### PR TITLE
Added strong naming to projects to 3.x

### PR DESCRIPTION
**Pull request by ramonsmits**

### CODE REVIEW

File: src/NServiceBus.Metrics.ServiceControl.Msmq/NServiceBus.Metrics.ServiceControl.Msmq.csproj [hunk 1]
```diff
@@ -8,6 +8,9 @@
     <GenerateAssemblyInfo>false</GenerateAssemblyInfo>
     <DebugType>full</DebugType>
     <DebugSymbols>true</DebugSymbols>
+    <AssemblyOriginatorKeyFile>$(SolutionDir)NServiceBus.snk</AssemblyOriginatorKeyFile>
+    <SignAssembly>true</SignAssembly>
+    <DelaySign>false</DelaySign>
```
Comment (bording) [inline src/NServiceBus.Metrics.ServiceControl.Msmq/NServiceBus.Metrics.ServiceControl.Msmq.csproj]: You don't need this.

File: src/NServiceBus.Metrics.ServiceControl.Msmq/NServiceBus.Metrics.ServiceControl.Msmq.csproj [hunk 2]
```diff
@@ -8,6 +8,9 @@
     <GenerateAssemblyInfo>false</GenerateAssemblyInfo>
     <DebugType>full</DebugType>
     <DebugSymbols>true</DebugSymbols>
+    <AssemblyOriginatorKeyFile>$(SolutionDir)NServiceBus.snk</AssemblyOriginatorKeyFile>
```
Comment (bording) [inline src/NServiceBus.Metrics.ServiceControl.Msmq/NServiceBus.Metrics.ServiceControl.Msmq.csproj]: I'd prefer to see things in the following order:

```
<SignAssembly>true</SignAssembly>
<AssemblyOriginatorKeyFile>..\NServiceBus.snk</AssemblyOriginatorKeyFile>
```

Which matches the order in the project files in other repos.

File: src/NServiceBus.Metrics.ServiceControl..Msmq.Tests/NServiceBus.Metrics.ServiceControl.Msmq.Tests.csproj [hunk 3]
```diff
@@ -2,6 +2,8 @@
 
   <PropertyGroup>
     <TargetFramework>net452</TargetFramework>
+    <AssemblyOriginatorKeyFile>$(SolutionDir)Test.snk</AssemblyOriginatorKeyFile>
```
Comment (bording) [inline src/NServiceBus.Metrics.ServiceControl..Msmq.Tests/NServiceBus.Metrics.ServiceControl.Msmq.Tests.csproj]: This should be 
```
<SignAssembly>true</SignAssembly>
<AssemblyOriginatorKeyFile>..\NServiceBus.snk</AssemblyOriginatorKeyFile>
```
Don't add a new key, just use the standard one.

**Comment by bording** [comment]
This should definitely not be merged until after #6 is fully merged back to develop. Right now you're likely only getting a warning that would otherwise be an error when the changes in #6 are included.

**Comment by ramonsmits** [comment]
@bording This is the compiler output:
```
1>------ Rebuild All started: Project: NServiceBus.Metrics.ServiceControl.Msmq, Configuration: Debug Any CPU ------
1>CSC : warning CS8002: Referenced assembly 'NServiceBus.Metrics.ServiceControl, Version=2.0.0.0, Culture=neutral, PublicKeyToken=null' does not have a strong name.
1>NServiceBus.Metrics.ServiceControl.Msmq -> C:\Users\Ramon\src\particular\NServiceBus.Metrics.ServiceControl.Msmq\binaries\net452\NServiceBus.Metrics.ServiceControl.Msmq.dll
1>Done building project "NServiceBus.Metrics.ServiceControl.Msmq.csproj".
```

I verified with `sn -vf NServiceBus.Metrics.ServiceControl.Msmq.dll` if the resulting assembly has a strong name and it does:
```
Microsoft (R) .NET Framework Strong Name Utility  Version 4.0.30319.0
Copyright (c) Microsoft Corporation.  All rights reserved.

Assembly 'NServiceBus.Metrics.ServiceControl.Msmq.dll' is valid
```

Just in case I also checked `NServiceBus.Metrics.ServiceControl.dll` which has a different output:
```
Microsoft (R) .NET Framework Strong Name Utility  Version 4.0.30319.0
Copyright (c) Microsoft Corporation.  All rights reserved.

NServiceBus.Metrics.ServiceControl.dll does not represent a strongly named assembly
```

**Comment by ramonsmits** [comment]
@bording I don't understand why it states the NServiceBus.snk is different, I copied it from the NServiceBus repo.

**Comment by ramonsmits** [comment]
Found the reason why it now compiles, but with a warning:

> Referencing unsigned assembly from signed is ok on some platforms (like CoreCLR). The compiler doesn't know what platform are you going to run on hence it no longer reports an error. We still report a warning since you might run into problems as you did on Desktop CLR.

**Pull request merged**